core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,12 +135,23 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var _deleteElementIfEmpty = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
+            _deleteElementIfEmpty();
```

Comment:
The _deleteElementIfEmpty() is needed here but in showAddItemButton it is needed to automatically delete the empty fields. Removing it from here